README作成

sushosushi opened the pull request:
#WHAT
READMEにデータベース設計記述

#WHY
chatspaceのデータベース設計のため

Commit:
Database Design

The time being

Commit:
correct README.md

Commit:
Edit README.md

Edit the GroupTable

mkoma414 commented on README.md:
```diff
@@ -22,3 +22,70 @@ Things you may want to cover:
+|name|string|null: false|
+|password|string|null: false|
+|password_confirm|string|null: false|
+
```
idは自動で振られるものなので記載は不要です。他のテーブルも同様です。

mkoma414 commented on README.md:
```diff
@@ -22,3 +22,70 @@ Things you may want to cover:
+|------|----|-------|
+|id|integer|null: false|
+|name|string|null: false|
+|password|string|null: false|
```
passwordはdeviseを使うと自動で作成されるので必ずしも書かなくて大丈夫です。記載する場合はemailなど他のカラムについても漏れなくご記入ください。

mkoma414 commented on README.md:
```diff
@@ -22,3 +22,70 @@ Things you may want to cover:
+### Association
+- has_many :groups,througth: :group_users
+- has_many :groups_users
+
```
・上記のアソシエーションの記述に誤りがあります。再度ご確認ください！
・メッセージとのアソシエーションが必要です。

mkoma414 commented on README.md:
```diff
@@ -22,3 +22,70 @@ Things you may want to cover:
+|------|----|-------|
+|id|integer|null: false|
+|name|string|null: false|
+
```
グループ名は重複しないようなカラム制約をかけてください。

mkoma414 commented on README.md:
```diff
@@ -22,3 +22,70 @@ Things you may want to cover:
+- has_many :users, through: :group_users
+- has_many :group_users
+- accepts_nested_attributes_for :group_users
+
```
・アソシエーションの記述を再度ご確認ください。
・accepts〜 の記述はアソシエーションではないので、ここには不要です。

mkoma414 commented on README.md:
```diff
@@ -22,3 +22,70 @@ Things you may want to cover:
+### Association
+- belongs_to :group
+- belongs_to :user
+
```
membersテーブルとgroup_userテーブルは同じ役割のはずなので
いずれか１つで大丈夫です。

mkoma414 commented:
最初から記述してあるreadmeの記述は不要ですので削除してください。

Commit:
README修正

#READMEを削除
group_userテーブル削除

taketnk commented on README.md:
```diff
@@ -1,24 +1,55 @@
+|Column|Type|Options|
+|------|----|-------|
+|group_name|string|null: false|
+|user_id|integer|null: false|
```
user_idは不要ですね。

taketnk commented on README.md:
```diff
@@ -1,24 +1,55 @@
+
+|Column|Type|Options|
+|------|----|-------|
+|group_name|string|null: false|
```
group_nameでもいいですが、groupsテーブル内なので、nameだけでもよいかと。

taketnk commented on README.md:
```diff
@@ -1,24 +1,55 @@
-* Database initialization
+|Column|Type|Options|
+|------|----|-------|
+|id|integer|null: false, unique: true|
```
idはデフォルトで付与されるため、書かなくても大丈夫です。

taketnk commented on README.md:
```diff
@@ -1,24 +1,55 @@
-* Configuration
 
-* Database creation
+## userテーブル
```
細かいですがusers(複数)テーブルでお願いします。

taketnk commented on README.md:
```diff
@@ -1,24 +1,55 @@
+
+
+
+## groupテーブル
```
groups(複数形)テーブルで

taketnk commented on README.md:
```diff
@@ -1,24 +1,55 @@
+- has_many :users, through: :members
+- has_many :members
+
+## messageテーブル
```
messagesテーブルで

taketnk commented on README.md:
```diff
@@ -1,24 +1,55 @@
+### Association
+- has_many :groups,througth: :members
+- has_many :members
+
```
userはメッセージを送信できるので、messagesテーブルに対する関係も記述してあげましょう。

taketnk commented on README.md:
```diff
@@ -1,24 +1,55 @@
+
+### Association
+- has_many :users, through: :members
+- has_many :members
```
groupには、たくさんのmessageが投稿されるので、messagesに対する関係も追記しておきましょう。

taketnk commented on README.md:
```diff
@@ -1,24 +1,55 @@
+|Column|Type|Options|
+|------|----|-------|
+|id|integer|null: false, unique: true|
+|name|string|null: false, unique: true|
```
nameカラムには、group作成機能を実装する際に、nameに対して、検索をかけるので、高速化するためにもindexを貼っておきましょう。

Commit:
README修正

・テーブル名を複数形に修正

・groups,usersテーブルのアソシエーションを修正

・usersテーブルにインデックス設定追加

・id,user_idを削除

・group_name→nameに修正

Commit:
README修正

・テーブル名を複数形に修正

・groups,usersテーブルのアソシエーションを修正

・usersテーブルにインデックス設定追加

・id,user_idを削除

・group_name→nameに修正
